server/src/main/java/org/opensearch/search/aggregations/bucket/terms/IncludeExclude.java

```diff
@@ -584,26 +577,54 @@ public boolean isPartitionBased() {
         return incNumPartitions > 0;
     }
 
-    private Automaton toAutomaton() {
-        Automaton a = null;
+    private Automaton toAutomaton(@Nullable IndexSettings indexSettings) {
+        int maxRegexLength = indexSettings == null ? -1 : indexSettings.getMaxRegexLength();
```

Comment:
When can index settings be null? Seems like this would result in an odd exception message of "...the allowed maximum of [-1]...".

Reply:
Oops, good catch. This condition should really be:

```
if (maxRegexLength > 0 && include.length() > maxRegexLength) {
```

I'll make this change

Reply:
https://github.com/opensearch-project/OpenSearch/pull/2814